lib/metasploit/framework/compiler/windows.rb

```diff
@@ -44,35 +44,47 @@ def self.compile_c_to_file(out_file, c_template, type=:exe, cpu=Metasm::Ia32.new
           File.write(out_file, pe)
         end
 
-        # Returns the binary of a randomized and compiled source code.
+        # Returns randomized c source code.
         #
         # @param c_template [String]
         # 
         # @raise [NotImplementedError] If the type is not supported.
         # @return [String] The compiled code.
-        def self.compile_random_c(c_template, opts={})
-          type = opts[:type] || :exe
-          cpu = opts[:cpu] || Metasm::Ia32.new
+        def self.generate_random_c(c_template, opts={})
           weight = opts[:weight] || 80
           headers = Compiler::Headers::Windows.new
           source_code = Compiler::Utils.normalize_code(c_template, headers)
+
           randomizer = Metasploit::Framework::Obfuscation::CRandomizer::Parser.new(weight)
           randomized_code = randomizer.parse(source_code)
-          self.compile_c(randomized_code.to_s, type, cpu)
+          randomized_code.to_s
+        end
+
+        # Returns the binary of a randomized and compiled source code.
+        #
+        # @param rand_c_template [String]
+        # 
+        # @raise [NotImplementedError] If the type is not supported.
+        # @return [String] The compiled code.
+        def self.compile_random_c(rand_c_template, opts={})
```

Comment:
This method looks redundant since it just invokes `.compile_c` with the default arguments and no longer does any randomization. This should be removed or it should invoke the new method to keep compatibility with the old API.